guacamole-ext/src/main/java/org/apache/guacamole/net/event/listener/Listener.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.guacamole.net.event.listener;
+
+/**
+ * A marker interface extended by all listener types. This interface is used
```

Comment:
I would fall on the "marker interface is an anti-pattern" side of the fence. If there is no shared functional interface between these listeners, then we shouldn't be defining a contract which states that there is.

That said, if it feels wrong to not have some common ancestor, then perhaps there *should* be a shared functional interface? At the moment, the only barrier I see preventing a common function for handling events is the variation on whether a handler is `void` or `boolean`, but there may be a better approach than the way things were done in the early guac days. Doing so would simplify the way these events get fired, and would avoid all the `instanceof` checks and casting.